packages/teleport/src/useUrlFiltering.tsx

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2021 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useMemo } from 'react';
+import { useLocation, useHistory } from 'react-router';
+import {
+  Label,
+  LabelTag,
+  LabelOption as Option,
+  makeLabelTag,
+} from 'teleport/services/resources';
+
+// PARAM_FILTER is a url query parameter for filters.
+const PARAM_FILTER = '?filter=';
+
+// FILTER_TYPE_LABEL defines a filter type for labels in a url query param.
+const FILTER_TYPE_LABEL = 'label:';
+
+export default function useUrlFiltering<T extends Filterable>(data: Array<T>) {
+  const history = useHistory();
+  const { search, pathname } = useLocation();
+  const selectedLabels = useMemo<Option[]>(() => getLabelsFromUrl(search), [
+    search,
+  ]);
+
+  const filteredData = useMemo(() => filterData(data, selectedLabels), [
+    data,
+    selectedLabels,
+  ]);
+
+  function apply(labels: Option[]) {
```

Comment:
Maybe `applyLabels`?

Reply:
i renamed `useUrlFiltering` to `useUrlLabelFiltering` so the hook name should imply its for labels